Src/DataTypes/S3.cs

```diff
@@ -1,3 +1,163 @@
-namespace GpkgMerger
+using System;
+using Amazon.S3;
+using Amazon.Runtime;
+using System.Collections.Generic;
+using GpkgMerger.Src.Batching;
+using Amazon.S3.Model;
+using GpkgMerger.Src.Utils;
+
+namespace GpkgMerger.Src.DataTypes
 {
+    public class S3 : Data
+    {
+        private AmazonS3Client client;
+        private string bucket;
+        private string continuationToken;
+        private bool endOfRead;
+
+        public S3(string serviceUrl, string bucket, string path, int batchSize) : base(DataType.S3, path, batchSize)
+        {
+            this.bucket = bucket;
+            this.continuationToken = null;
+            this.endOfRead = false;
+
+            // Get s3 credentials
+            string accessKey = Environment.GetEnvironmentVariable("S3_ACCESS_KEY");
+            string secretKey = Environment.GetEnvironmentVariable("S3_SECRET_KEY");
+            var credentials = new BasicAWSCredentials(accessKey, secretKey);
+            var config = new AmazonS3Config
+            {
+                RegionEndpoint = Amazon.RegionEndpoint.USEast1,
+                ServiceURL = serviceUrl,
+                UseHttp = true,
```

Comment:
does that support both http and https endpoints?

Reply:
Apparently it has no effect. The url needs to include the http or https anyway... Therefore this will be removed.